Src/FluentAssertions/Primitives/BooleanAssertions.cs

```diff
@@ -118,6 +118,16 @@ public AndConstraint<TAssertions> NotBe(bool unexpected, string because = "", pa
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+    public AndConstraint<TAssertions> Imply(bool expectation, string because = "", params object[] becauseArgs)
+    {
+        Execute.Assertion            
+            .ForCondition(!Subject.Value || expectation)
+            .BecauseOf(because, becauseArgs)
+            .FailWith("Expected {context:boolean} ({0}) to imply {1} ({2}){reason}, but it did not.", Subject, nameof(expectation), expectation);
```

Comment:
I am not very happy with this solution, but currently the only possible solution..

Maybe we should keep this in mind until `[CallerArgumentExpression]`

Reply:
How about
`"Expected {context:boolean} to imply {0}{reason}, but found {1}.", expectation, Subject`

Reply:
Hmm.. This would result in e.g. `"Expected subject to imply True, but found False.", expectation, Subject`

This does not help IMO.